reader.go

```diff
@@ -261,3 +261,20 @@ func (r *Reader) WriteTo(w io.Writer) (n int64, err error) {
 		}
 	}
 }
+
+// CheckLZ4 returns a bool indicating if the first 32 byets mactch LZ4 header
+// from the io.Reader stream.
+//
+// False if it could not determine, e.g. error in reading from io.Reader.
+//
+// Don't pass original io.Reader for checking, otherwise internal read pos of
+// the io.Reader will be off by 32, causing a reader error. Instead, consider
+// pre-read bytes into a slice for archive checking, and then io.MultiReader
+// to concatenate it for the real lz4 reading.
+func CheckLZ4(r io.Reader) bool {
+	f := lz4stream.NewFrame()
+	if err := f.ParseHeaders(r); err != nil {
+		return false
+	}
```

Comment:
This swallows all errors from the Reader. I would expect to be able to get those errors returned, so my program can distinguish I/O errors from "not LZ4". (Network errors in particular may not be repeatable.)

Reply:
Done.

I don't mind exposing the error. Just note, in case a user want to check the error, he/she need the bad magic number error exposed at lz4 level, than hidden within `internal/lz4errors`,  e.g. `ErrInvalidFrame                  Error = "lz4: bad magic number"`